### PR TITLE
mon: resolve warning about inconsistent variable types

### PR DESCRIPTION
**Pull request by EdwardVitor**
resolve a warning issue about inconsistent variable types with Building CXX object src/mon/CMakeFiles/mon.dir/Elector.cc.o ../src/mon/Elector.cc: In member function 'void Elector::notify_rank_removed(int)': ../src/mon/Elector.cc:765:15: warning: comparison of integer expressions of different signedness: 'int' and 'unsigned int' [-Wsign-compare]
   if (rank_removed < paxos_size()) {

Fixes: https://tracker.ceph.com/issues/63917

Signed-off-by: cuiming <cuiming_yewu@cmss.chinamobile.com>

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

**Comment by EdwardVitor** [comment]
@rzarzynski @cbodley @markhpc @zdover23 please help review,thanks

File: src/mon/Elector.cc [hunk 1]
```diff
@@ -762,7 +762,7 @@ void Elector::notify_rank_removed(unsigned rank_removed, unsigned new_rank)
      In the case where we are removing the highest rank,
      we erase the removed rank from all sets.
    */
-  if (rank_removed < paxos_size()) {
+  if (static_cast<unsigned int>(rank_removed) < paxos_size()) {
```
Comment (cbodley) [inline src/mon/Elector.cc:765]: consider using the c++20 function [cmp_less()](https://en.cppreference.com/w/cpp/utility/intcmp) which handles signed/unsigned comparisons transparently

**Review by batrick — CHANGES_REQUESTED** [review]
Please create a [tracker issue](https://tracker.ceph.com/) so that this issue can be tracked for backporting.

Then, [please annotate the commit which fixes/resolves a Ceph tracker issue with](https://github.com/ceph/ceph/blob/master/SubmittingPatches.rst#fixes-lines):

`Fixes: http://tracker.ceph.com/issues/...`

This is essential when examining the history of the repository (this commit fixes what) and helps merge scripts identify issues that have been resolved by a merge. See [this article](https://help.github.com/en/articles/changing-a-commit-message) on GitHub on how to amend commits and update your pull request.

File: src/mon/Elector.cc (hunk 1 above)
Comment (EdwardVitor) [reply src/mon/Elector.cc]: Try using this approach to submit the new changes.

**Review by cbodley — APPROVED** [review]

**Comment by EdwardVitor** [comment]
@cbodley Is there anything else that needs to be modified in this PR

**Comment by cbodley** [comment]
> @cbodley Is there anything else that needs to be modified in this PR

hey @EdwardVitor, it looks fine to me but @batrick requested changes above. i created https://tracker.ceph.com/issues/63917 to track backports to reef and quincy since https://github.com/ceph/ceph/pull/46029 was introduced in reef and backported from there (cc @kamoltat)

can you please update your commit message to add the line `Fixes: https://tracker.ceph.com/issues/63917`? once that's addressed, we'll add the `needs-qa` tag to get it tested

**Comment by EdwardVitor** [comment]
> > @cbodley Is there anything else that needs to be modified in this PR
> 
> hey @EdwardVitor, it looks fine to me but @batrick requested changes above. i created https://tracker.ceph.com/issues/63917 to track backports to reef and quincy since #46029 was introduced in reef and backported from there (cc @kamoltat)
> 
> can you please update your commit message to add the line `Fixes: https://tracker.ceph.com/issues/63917`? once that's addressed, we'll add the `needs-qa` tag to get it tested

OK, thanks !

**Comment by cbodley** [comment]
jenkins test api

**Review by batrick — APPROVED** [review]

**Comment by kamoltat** [comment]
RADOS APPROVED 
https://tracker.ceph.com/projects/rados/wiki/MAIN#httpstrellocomcD0n65np11923-wip-yuri-testing-2024-01-09-1331

**Pull request merged**